daemon/policy.go

```diff
@@ -269,17 +232,19 @@ func (d *Daemon) PolicyAdd(rules policyAPI.Rules, opts *AddOptions) (uint64, err
 		return d.policy.GetRevision(), err
 	}
 
-	rev, err := d.policyAdd(rules, opts, prefixes)
-	if err != nil {
-		// Don't leak identities allocated above.
-		_ = d.prefixLengths.Delete(prefixes)
-		if err2 := ipcache.ReleaseCIDRs(prefixes); err2 != nil {
-			log.WithError(err2).WithField("prefixes", prefixes).Warn(
-				"Failed to release CIDRs during policy import failure")
+	d.policy.Mutex.Lock()
+	if opts != nil && opts.Replace {
+		// Make copy of rules matching labels of new rules while
```

Comment:
Nit: This comment doesn't seem to match the code anymore, because we aren't saving any old rules any more.